components/kubeflow-pipeline/pipeline_components/analytics-server/flwr_analytics_server/fivenum.py

```diff
@@ -0,0 +1,68 @@
+""" 
+
+Five number summary given histogram.   
+Bases for a Boxplot.  
+
+Katulu GmbH  
+(c) 2022  
+
+"""
+
+import numpy as np
+from dataclasses import dataclass
+from typing import List, Optional
+from .fedhist import HistogramData
+
+
+@dataclass
+class fiveNum:
+    """Five numbers for box plot."""
+
+    minimum: float
+    quartile_first: float
+    median: float
+    quartile_third: float
+    maximum: float
+
+
+@dataclass
+class fiveNumSummary:
+    """Five number summary."""
+
+    histogram: HistogramData
+    hmin: float
+    hmax: float
+    fivenum: Optional[fiveNum] = None
```

Comment:
Let's use data classes to group similar parameters, not to group functionality related to five number summary, as that's already captured by having this package.
With this current approach, it's easy to overlook which of the parameters are input values and expected to set when creating an instance of this class and which values will be set when calling `compute`. Instead, let's have a single function `compute_fivesum` and a helper function `inverse_value_bin`. By having:

```python
def compute_fivesum(hmin: float, hmax: float, histogram: HistogramData) -> FiveNum
```

it is clear which values are input and what is being output by the function. As there's only 3 input parameters, it's fine to name them directly, though once we have more parameters, it makes sense to create a separate data class to capture all input values like all output which are captured with the `fiveNum` class.

Reply:
Make sense. Updated.